packages/hosted-form-v2/src/hosted-field.ts

```diff
@@ -0,0 +1,186 @@
+        private _placeholder: string,
+        private _accessibilityLabel: string,
+        private _styles: HostedFieldStylesMap,
+        private _eventPoster: IframeEventPoster<HostedFieldEvent>,
+        private _eventListener: IframeEventListener<HostedInputEventMap>,
+        private _detachmentObserver: DetachmentObserver,
+    ) {
+        this._iframe = document.createElement('iframe');
+
+        this._iframe.src = `/checkout/payment/hosted-field?version=${LIBRARY_VERSION}`;
+        this._iframe.style.border = 'none';
+        this._iframe.style.height = '100%';
+        this._iframe.style.overflow = 'hidden';
+        this._iframe.style.width = '100%';
+    }
+
+    getType(): HostedFieldType {
+        return this._type;
+    }
+
+    async attach(): Promise<void> {
+        const container = document.getElementById(this._containerId);
+
+        if (!container) {
+            throw new InvalidHostedFormConfigError(
+                'Unable to proceed because the provided container ID is not valid.',
+            );
+        }
+
+        container.appendChild(this._iframe);
+        this._eventListener.listen();
+
+        const promise = fromEvent(this._iframe, 'load')
+            .pipe(
+                switchMap(async ({ target }) => {
+                    const contentWindow = target && (target as HTMLIFrameElement).contentWindow;
+
+                    if (!contentWindow) {
+                        throw new Error('The content window of the iframe cannot be accessed.');
+                    }
+
+                    this._eventPoster.setTarget(contentWindow);
+
+                    await this._eventPoster.post(
+                        {
+                            type: HostedFieldEventType.AttachRequested,
+                            payload: {
+                                accessibilityLabel: this._accessibilityLabel,
+                                fontUrls: this._getFontUrls(),
+                                placeholder: this._placeholder,
+                                styles: this._styles,
+                                origin: document.location.origin,
+                                type: this._type,
+                            },
+                        },
+                        {
+                            successType: HostedInputEventType.AttachSucceeded,
+                            errorType: HostedInputEventType.AttachFailed,
+                        },
+                    );
+                }),
+                take(1),
+            )
+            .toPromise();
+
+        await this._detachmentObserver.ensurePresence([this._iframe], promise);
+    }
+
+    detach(): void {
+        if (!this._iframe.parentElement) {
+            return;
+        }
+
+        this._iframe.parentElement.removeChild(this._iframe);
+        this._eventListener.stopListen();
+    }
+
+    async submitForm(
+        fields: HostedFieldType[],
+        data: HostedFormOrderData,
```

Comment:
It depends what `HostedFormOrderData` will eventually contain. But if it varies depending on the context (checkout vs manual orders vs my account), it'd be better if we're using generic rather than a single interface.

Do you know if there's any data that needs to be submitted from the host window to BigPay?

Reply:
Makes sense, can use generic type here.

I am not sure about what data needs to be submitted for manual-orders. @bc-peng Do you know?

Reply:
For the payload, we follow the new payment endpoint's YAML file.
https://github.com/bigcommerce/api-schema/blob/26a229103c0a285dea3e1850595814fe655e090a/bigpay/payments.yaml#L37
If we resue this method to submit all payment methods' payment, the `data` type will have many variations.

Reply:
@davidchin We will possibly need a new method to submit manual order payment and the method for existing payments can stay same, so I guess the type can be as it is in the core package for `checkoutjs` and `storefront-account-payments`. Can bring it over when we work on changes for using this package in those places.  
For manual order, the new submit method will be implemented as a part of [CHECKOUT-8275](https://bigcommercecloud.atlassian.net/browse/CHECKOUT-8275)

Reply:
Ok, in that case, I will review the code around payment submission in the follow-up PR. 🙂